MetaMask/metamask-extension
# Added wallet_revokePermissions rpc method

## **Description**
This PR adds `wallet_revokePermissions` rpc method support that was added in `permission-controller/rpc-methods` that went in this PR https://github.com/MetaMask/core/pull/1889.

All that is needed is to add the hook as the rpc-method gets automatically added [here](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/rpc-method-middleware/createMethodMiddleware.js#L13).


## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1681

## **Manual testing steps**

1. Go to this page https://docs.metamask.io/wallet/reference/wallet_requestpermissions/
2. Hit `Send Request` to request permissions
3. Click Next, Click Connect
4. Go to this page https://docs.metamask.io/wallet/reference/wallet_getpermissions/
5. Click `Send Request` to get the permissions
6. See that there is permissions
7. paste this in the console:
```js
await window.ethereum.request({
  "method": "wallet_revokePermissions",
  "params": [
    {
      "eth_accounts": {}
    }
  ]
});
```
8. Click `Send Request` to get the permissions
9. see that there is no permissions and they have been revoked.

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
